src/Operators/vorticity_operators.jl

```diff
@@ -18,25 +32,9 @@
 
  The vertical circulation associated with horizontal velocities ``u`` and ``v``.
  """
-@inline function Γᶠᶠᶜ(i, j, k, grid::OrthogonalSphericalShellGrid, u, v)
-    # South-west corner
-    if i == 1 && j == 1
-        return Δy_qᶜᶠᶜ(i, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
-
-    # South-east corner
-    elseif i == grid.Nx+1 && j == 1
-        return - Δy_qᶜᶠᶜ(i-1, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
-
-    # North-west corner
-    elseif i == 1 && j == grid.Ny+1
-        return Δy_qᶜᶠᶜ(i, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
-
-    # North-east corner
-    elseif i == grid.Nx+1 && j == grid.Ny+1
-        return - Δy_qᶜᶠᶜ(i-1, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u)
-
-    # Not a corner
-    else
-        return δxᶠᵃᵃ(i, j, k, grid, Δy_qᶜᶠᶜ, v) - δyᵃᶠᵃ(i, j, k, grid, Δx_qᶠᶜᶜ, u)
-    end
-end
+@inline Γᶠᶠᶜ(i, j, k, grid::OrthogonalSphericalShellGrid, u, v) = 
+    ifelse(on_south_west_corner(i, j, grid) | on_north_west_corner(i, j, grid),
+           Δy_qᶜᶠᶜ(i, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u), 
+           ifelse(on_south_east_corner(i, j, grid) | on_north_east_corner(i, j, grid),
+                  - Δy_qᶜᶠᶜ(i-1, j, k, grid, v) - Δx_qᶠᶜᶜ(i, j, k, grid, u) + Δx_qᶠᶜᶜ(i, j-1, k, grid, u), 
+                  δxᶠᵃᵃ(i, j, k, grid, Δy_qᶜᶠᶜ, v) - δyᵃᶠᵃ(i, j, k, grid, Δx_qᶠᶜᶜ, u)))
```

Comment:
are we sure this is the indented behavior we want?

```Julia
julia> a = 1
1

julia> ifelse(a == 1 | a ==2, 10, 20)
20

julia> ifelse(a == 1 || a ==2, 10, 20)
10
```

Do we want `|` or `||`?

Reply:
Seems that you assume that
```Julia
julia> 1 == 1 | 1 == 2
```
returns true. But it does not!

Reply:
You want 

```julia
(1 == 1) | (1 == 2)
```

indeed, operator precedence rules treat `|` and `||` differently.

Note that `||` is "short-circuiting" (bad for GPU performance in a kernel, but ensures that the "dead" branch does not execute at all) whereas `|` is not (but both branches execute)

Reply:
Gotcha!